bundles/CoreBundle/src/Migrations/Version20221220152444.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Pimcore\Db;
+use Pimcore\Model\Tool\SettingsStore;
+
+/**
+ * Set installed if not set yet
+ */
+final class Version20221220152444 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Checks if Web2Print is installed in Settingsstore';
+    }
+
+    public function up(Schema $schema): void
+    {
+        if(!SettingsStore::get('BUNDLE_INSTALLED__Pimcore\\Bundle\\WebToPrintBundle\\PimcoreWebToPrintBundle', 'pimcore')) {
+            SettingsStore::set('BUNDLE_INSTALLED__Pimcore\\Bundle\\WebToPrintBundle\\PimcoreWebToPrintBundle', true, 'bool', 'pimcore');
```

Comment:
At least for `up`, something like
```suggestion
            SettingsStore::set('BUNDLE_INSTALLED__Pimcore\\Bundle\\WebToPrintBundle\\PimcoreWebToPrintBundle', $web_to_print_config['enableInDefaultView'], 'bool', 'pimcore');
```
if it was never activated before, on migration, it should stay deactivated.

For down, we don't know anymore, but, we could roughly check if there's any existing printpage? that could give an hint that it was previously checked, even though one could have not cleaned up any of those and still have web2print disabled.

Reply:
This could potentially break the installer, since the permissions are already there.
If you upgrade to 11.x and then try to install the WebToPrintBundle (when it was not activated before) will break the installation since the permission is already in the table. 
We could check the `enableInDefaultView` and if it is false we also remove the permission?

Reply:
ah, true, maybe we should change the installer to be more idempotent, instead of `insert`, to be an `upsert`.

The migration `up` here also need to consider if the `enableInDefaultView` is false and in that case, it should also remove the enum from `documents.types` and `documents_printpage`

Reply:
Hm, I am not so sure if we should do this. We cannot actual know if someone used this feature but disabled it for now. We would delete data with this migration, just because one setting was not set to true. This does not feel right to me.

Reply:
We talked in the daily about it. We should not delete anything in the migration. Just do not mark it as installed and make the installer idempotent. We cannot trust this flag enough.